### ISSUE TITLE
BUG: lots of p values are zeros after mulitple correction using default parameters

### ISSUE THREAD

**Issue by hsiaoyi0504** [statsmodels/statsmodels#7465]
Hi there, this is more a FAQ than a bug report.
When I am using the `statsmodels.stats.multitest.multipletests` (default parameter) on my t-test results, I  get lots of 0s for p-value (see following example). Is it something expected? Should I try to use other multiple correction methods?
It's hard to plot them on a 2-d plot with fold-changes.

After correction   Before correction
0.000000e+00    5.500356e-20   
0.000000e+00    3.334809e-40   
0.000000e+00    3.702405e-36      
0.000000e+00    3.395937e-20
0.000000e+00    2.425107e-30

**Comment by josef-pkt** [comment]
default is holm-sidak.
sidak uses `1 - pvalue` which will be just 1 in those cases. pvalues are smaller than floating point precision at 1.

`holm` uses Bonferroni correction in the intermediat steps and should not have that problem AFAICS.

So this is a consequence of how how sidak is computed using floating point precision.
Statistically it's not really relevant if pvalue is 1e-20 or 1e-40 or 0.

What's the scaling on your plot? Can you just clip corrected pvalues at 1e-20 (to have nonzero lower bound)?

AFAIR, pvalue correction should never lower the pvalue, So we could add a bound in there,
Maybe it's possible to rewrite Sidak using log1p, expm1 to gain precision in this case.

**Comment by josef-pkt** [comment]
How many pvalues do you have?

It might not be only the precision problems coming from very small p-values, but also from taking power with number of tests.

**Comment by hsiaoyi0504** [comment]
I use -log p value for the plot. In my case, I have 6698 p values.

**Comment by josef-pkt** [comment]
looks like expm1 log1p works for very small pvalues

```
pvals, ntests = 1e-40, 6000
-np.expm1(6000*np.log1p(-pvals)), 1 - np.power((1. - pvals), ntests), np.power((1. - pvals), ntests)
```

**Comment by hsiaoyi0504** [comment]
Should I try to submit a pull request?

**Comment by josef-pkt** [comment]
pull request would be good. Only needs a simple unit test for a case that is currently zero, regular cases will be covered by current unit tests

AFAICS, all cases that use Sidak  `1 - np.power((1. - pvals), ntests)` can be replaced by the exp1m/log1p version.
There is one duplicated computation of it in "holm-sidak" where we can remove one.

**Comment by josef-pkt** [comment]
we can leave one of the `1 - np.power((1. - pvals),  np.arange(ntests, 0, -1))` in a code comment. It's much easier to read and understand than the exp1m/log1p version

**Comment by josef-pkt** [comment]
removing FAQ label, this is not a corner solution
precision problem fixed in PR #7668 based pm original PR #7466
